src/search/merge_and_shrink/merge_scoring_function_miasm.h

```diff
@@ -12,7 +12,7 @@ namespace merge_and_shrink {
 class ShrinkStrategy;
 class MergeScoringFunctionMIASM : public MergeScoringFunction {
     const bool use_caching;
-    std::shared_ptr<ShrinkStrategy> shrink_strategy;
+    const std::shared_ptr<ShrinkStrategy> &shrink_strategy;
```

Comment:
Okay I figured out why i did change something here.
clang-tidy:
```Errors and warnings:
/downward/src/search/merge_and_shrink/merge_scoring_function_miasm.cc:24:23: 
warning: parameter 'shrink_strategy' is passed by value and only copied once;
consider moving it to avoid unnecessary copies [performance-unnecessary-value-param]
```

I did undo it here but changed the constructor to actually move the shrink_strategy

Reply:
For someone trying to understand the rationale of this, I think this is a good link:
 https://stackoverflow.com/questions/41871115/why-would-i-stdmove-an-stdshared-ptr

Salomé was right we definitely don't want to change the member from a pointer to a pointer reference; this is very likely to crash and burn once we actually try to access the shrink strategy later because we would reference a temporary value. More generally, I haven't looked at it much, but I don't think there should be many cases where we need to change the data members in the class definitions. The cases where we do make such changes we need to look at carefully.